client/components/SearchOperationList.js

```diff
@@ -155,7 +160,7 @@ export default class SearchComponent extends React.Component {
                 store.getOperationTypes().map((t) => <option key={t.id} value={t.id}>{store.operationTypeToLabel(t.id)}</option>)
             );
 
-            details = <div className="panel-body transition-expand">
+            details = <div className="panel-body transition-expand" key={Date.now()}>
```

Comment:
This will force to render a new DOM node. Without it the form fields would keep their value. For example if I select "Card" in the type dropdown, clearing the form would not reset the dropdown state.